### ISSUE TITLE
Don't display the RealName in the search menu

### ISSUE THREAD

**Issue by behemoth36** [RocketChat/Rocket.Chat.ReactNative#2653]
### Description:

When you search a user on mobile app by the search menu, it display the uid of the user not the realname. (The function use real name is enable on the server). It works only by the directory menu.
 
### Environment Information:

- Rocket.Chat Server Version: 3.8.0
- Rocket.Chat App Version: 4.11.0.2137
- Device Name: Android/MacOS
- OS Version: Android 7.0

### Steps to reproduce:

1. Touch search icon
2. Type a user name
3. It display uid

### Expected behavior:

The result of the research display by name not uid

### Actual behavior:

The result of the research display by uid not user name

### Additional context:

![mobile1](https://user-images.githubusercontent.com/71933767/99547534-ed2f8300-29b7-11eb-8ecb-a04be6306e29.png)

**Comment by markusstoll** [comment]
I do observe it even more strange: I do get search results presented duplicate - one with real name and one with user name field.

**Comment by Dharm3438** [comment]
I would be glad to take up this issue, can anyone please brief me on what has to be done here? Thanks.

**Comment by yash-rajpal** [comment]
@behemoth36 It works fine for me, I checked on release app version 4.12.1.2143 and also checked on debug mode. Please update your app and let us know.
![Screenshot_2020-12-17-01-49-48-425_chat rocket android](https://user-images.githubusercontent.com/58601732/102402221-7a431780-400a-11eb-97f1-b34ecb2a5097.jpg)

**Comment by markusstoll** [comment]
Hi, I am using latest Beta 4.13.0.22006 and its still not correct

I am search for first Name part which is uniq and should match ONE result, but I do get TWO results, one showing the real name and a second entry, showing the username.

**Comment by dossantg** [comment]
Hello,

It works if you have already started a direct coversation with a person.
On the other hand, if we search for a person for the first time it displays the username and not the real name.

We have the same behavior on Andoid and iOS on the latest versions available in the stores

**Comment by markusstoll** [comment]
Indeed, if I search for the first time for a given user, I have one result, showing the user name (not real name).

So the client searches separately (locally and using the search API) and merges the results without consolidation (neither the hits nor the presentation form). Is this observation correct?

**Comment by deepak0x** [comment]
Hey @diegolmello @Rohit3523 — went through the search and display logic and this looks resolved.

Search results in `search.ts` set `fname: user.name` (the real name) alongside `name: user.username`. Then `getRoomTitle()` in `helpers.ts` (line 59) returns `fname` when `useRealName` is true:

```
return ((room?.prid || useRealName) && room?.fname) || room?.name;
```

`RoomsListView` reads `UI_Use_Real_Name` from Redux and passes it through the component chain. So when the setting is enabled, search results display the real name. Think this can be closed.
